### DISCUSSION TITLE
Cypher - Where clause on internal id

### DISCUSSION THREAD

**Discussion by fedehann** [kuzudb/kuzu#3448] [Q&A] [answered]
Hi, 
is it possible to use a WHERE clause in a Cypher statement to filter by the internal ID of a node or a rel?

Something like:

```
MATCH(n)
where id(n) = '0:0' 
return n
```
?

[Edit] Of course, i could do:

```
MATCH(n)
where cast(id(n), 'string') = '0:0' 
return n
```

But that seems kind of horrible, and possibly slow.

**Comment by andyfengHKU** [comment, marked as answer]
Hi @fedehann, normally we don't wanna expose internal id for external querying. 

But you can query if with
```
MATCH(n) where label(n)=<label_name> AND offset(id(n)) = 0  return n
```
where `label(n)` returns  label name with STRING type and `offset(id(n))` returns  offset with INT64 type.


> But that seems kind of horrible, and possibly slow.

Yeah it will be slightly slower. But casting can be done fairly quick and will hardly become the bottleneck of a query
